### PR TITLE
Add - Automatic Release Post Utility

### PR DESCRIPTION
**Pull request by davidgarywood**
This adds a utility command line app, to support automating our release process.

* Reads the most recent release from our tags, and parses it to retrieve what has changed.
* Retrieves author information from the changes. Attempts to return either the author's linked Mastodon account from their Github user, or their GitHub username.

This, combined with a GH action and appropriate secrets, could be triggered when a new release is made

### CODE REVIEW

**Review by kkostov — APPROVED** [review]
Love it 😍!

File: Examples/tootsdk-release/Sources/tootsdk-release/Posts/MakePost.swift [hunk 1]
```diff
@@ -0,0 +1,179 @@
...
+        return unique
+    }
+    
+    private func getUserMastodon(user: String) async throws -> String? {
+        let mastodonURL = "https://api.github.com/users/" + user + "/social_accounts"
+        let (data, _ ) = try await urlSession.data(from: URL(string: mastodonURL)!)
+        
+        let socialAccounts = try JSONDecoder().decode([SocialAccount].self, from: data)
+        
+        if let userMastodonURL = socialAccounts.first(where: { $0.provider == "mastodon" })?.url {
```
Comment (kkostov) [inline Examples/tootsdk-release/Sources/tootsdk-release/Posts/MakePost.swift:128]: Note that Microsoft only supports mastodon, social links to other server types are not possible.

File: Examples/tootsdk-release/Sources/tootsdk-release/Posts/MakePost.swift [hunk 2]
```diff
@@ -0,0 +1,179 @@
...
+            }
+        case .failure(let error):
+            print(error.localizedDescription)
+        }
+        
+        return nil
+    }
+    
+    private func parseChanges(html: String) async throws -> [String] {
+        let changes = html.split(separator: "<li>")
```
Comment (kkostov) [inline Examples/tootsdk-release/Sources/tootsdk-release/Posts/MakePost.swift:73]: Is this based on the change log generated by GitHub?

Comment (davidgarywood) [reply Examples/tootsdk-release/Sources/tootsdk-release/Posts/MakePost.swift:73]: Yes !

File: Examples/tootsdk-release/Sources/tootsdk-release/Posts/MakePost.swift (hunk 1 above)
Comment (davidgarywood) [reply Examples/tootsdk-release/Sources/tootsdk-release/Posts/MakePost.swift:128]: Yeah. It’s frustrating but if they support others we can add to this

**Pull request merged**